simple insert_dict() builder

edvardm opened the pull request:
## Additions

- add helper `insert_dict()` for QueryBuilder instances
- guard to prevent use of `columns()` while using `insert_dict()`
- basic tests

Idea pretty much the same as in issue #533 :

```python
Query().into(self.table_abc).insert_dict({"num": 1, "timestamp": datetime(2023, 4, 18)})

# INSERT INTO "abc" ("num","timestamp") VALUES (1,'2023-04-18T00:00:00')
```

Given how it's implemented, maybe `from_dict()` would be more suitable, as `into()` is already part of context? Multi-value inserts are trickier though, as they seem to vary between engines

----

Edit:

- Simple example in README
- Just use class method `Query.into` in tests

wd60622 commented on pypika/tests/test_inserts.py:
```diff
@@ -168,6 +169,28 @@ def test_insert_with_statement(self):
+    table_abc = Table("abc")
+
+    def test_inserting_simple_dictionary(self):
+        q = Query().into(self.table_abc).insert_dict({"c1": "value", "c2": 1})
```
Any reason for not using `Query.into` since `into` is a classmethod?

wd60622 commented on pypika/tests/test_inserts.py:
```diff
@@ -168,6 +169,28 @@ def test_insert_with_statement(self):
+        self.assertEqual("INSERT INTO \"abc\" (\"num\",\"timestamp\") VALUES (1,'2023-04-18T00:00:00'),(2,'2023-04-19T00:00:00')", str(q))
+
+    def test_columns_is_not_allowed_with_insert_dict(self):
+        with self.assertRaises(QueryException):
+            Query().into(self.table_abc).columns("a", "b").insert_dict({"num": 1})
```
Any more to be checked about the error? Maybe regex match on the message?

i.e https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex

edvardm replied:
These are all good and valid points. Not working that much Python nowadays, but should be simple address. Thanks!

williambdean reviewed (commented):
Able to add this into the README in order to document?

edvardm committed:
simple insert_dict() builder

edvardm committed:
CR: just use class method (Query().into -> Query.into)

edvardm committed:
make insert_dict() api less confusing

in SQL builders, order of calling individual functions should
not matter. Originally it was allowed to first call columns()
and then insert_dict as long as the two matched, but I'm not even sure
it worked, as we should have used set comparison instead.

It is also better to have symmetric API in

edvardm committed:
insert_into() example to README

edvardm commented:
Sorry for delay, some quite tragic happened at work and forgot about this for some time. Hopefully resolved noted issues

Commit:
fix: README.rst (#801)

Commit:
run pre-commit hooks

AzisK commented:
Hey @edvardm. I am sorry I am looking at this only now. Is it still relevant?

edvardm commented:
NP, it was  just what I had found useful at the time in a small internal tool -> closing

AzisK commented:
It seems like a neat addition but up to you @edvardm
